src/content/en/fundamentals/engage-and-retain/push-notifications/display-a-notification.md

```diff
@@ -0,0 +1,407 @@
+project_path: /web/_project.yaml
+book_path: /web/fundamentals/_book.yaml
+
+{# wf_updated_on: 2017-03-03 #}
+{# wf_published_on: 2016-06-30 #}
+
+# Displaying a Notification {: .page-title }
+
+{% include "web/_shared/contributors/mattgaunt.html" %}
+
+
+
+I've split up notification options into two sections, one that deals with the visual aspects
+(this section) and one section that explains the behavioural aspects of notifications.
+
+The reason for this is that every developer will need to be worried about the visual aspects
+but the behavioural aspects will depend on your specific to your use case.
+
+All of the source code for these demo's is taken from a demo page I put together. If you want
+to test them out for yourself then click the button below.
+
+<a class="button" href="https://web-push-book.gauntface.com/demos/notification-examples/"
+target="\_blank">Notification Demos</a>
+
+## Visual Options
+
+The API for showing a notification is simply:
+
+    <ServiceWorkerRegistration>.showNotification(<title>, <options>);
+
+Where the title is a string and options can be any of the following:
+
+    {
+      "//": "Visual Options",
+      "body": "<String>",
+      "icon": "<URL String>",
+      "image": "<URL String>",
+      "badge": "<URL String>",
+      "vibrate": "<Array of Integers>",
+      "sound": "<URL String>",
+      "dir": "<String of 'auto' | 'ltr' | 'rtl'>",
+
+      "//": "Behavioural Options",
+      "tag": "<String>",
+      "data": "<Anything>",
+      "requireInteraction": "<boolean>",
+      "renotify": "<Boolean>",
+      "silent": "<Boolean>",
+
+      "//": "Both Visual & Behavioural Options",
+      "actions": "<Array of Strings>",
+
+      "//": "Doesn't seem to affect any visual aspect of the notification.",
+      "timestamp": "<Long>"
+    }
+
+First let's look at the visual options. For desktop and mobile, here are how the different
+options are used.
+
+![Dissection of the UI of a Notification](./images/notification-ui.png){: .center-image }
+
+
+### Title and Body Options
+
+The title and body options are exactly as they sound, two different pieces of text to display
+on the notification.
+
+If we ran the following code:
+
+        const title = 'Simple Title';
+        const options = {
+          body: 'Simple piece of body text.\nSecond line of body text :)'
+        };
+        registration.showNotification(title, options);
+
+We'd get this notification on Chrome:
+
+![Notification with Title and Body Text on Chrome on Linux.](./images/notification-screenshots/desktop/chrome-title-body.png){: .center-image }
+
+On Firefox on Linux it would look like this:
+
+![Notification with Title and Body Text on Firefox on Linux.](./images/notification-screenshots/desktop/firefox-title-body.png){: .center-image }
```

Comment:
This isn't a figure caption though, this is a fallback text / accessibility piece of text.